api/prometheus/v1/example_test.go

```diff
@@ -76,6 +78,118 @@ func ExampleAPI_queryRange() {
 	fmt.Printf("Result:\n%v\n", result)
 }
 
+type userAgentRoundTripper struct {
+	name string
+	rt   http.RoundTripper
+}
+
+// RoundTrip implements the http.RoundTripper interface.
+func (u userAgentRoundTripper) RoundTrip(r *http.Request) (*http.Response, error) {
+	if r.UserAgent() == "" {
+		// The specification of http.RoundTripper says that it shouldn't mutate
+		// the request so make a copy of req.Header since this is all that is
+		// modified.
+		r2 := new(http.Request)
+		*r2 = *r
+		r2.Header = make(http.Header)
+		for k, s := range r.Header {
+			r2.Header[k] = s
+		}
+		r2.Header.Set("User-Agent", u.name)
+		r = r2
+	}
+	return u.rt.RoundTrip(r)
+}
+
+func ExampleAPI_queryRange_withUserAgent() {
```

Comment:
It appears double-underscoring doesn't work. `ExampleAPI_queryRangeWithUserAgent` should work.
Same for the below.

Reply:
ack